Snapshot at 0724a.

soumkin opened the pull request:
以下を実施しました。
- 吸着機能
  - VacuumGripperSimulatorItem に limitCheckStartTime プロパティを追加した。
    
    シミュレーション開始からしばらくの間、dJointFeedback に設定される値が大きすぎるようなので、限界判定の開始を遅らせることができるようにした。
    
    デフォルトでは、開始から 0.3 秒までは限界判定を行わない。0.3秒という値は実験により求めた。
- ビス打ち機能
  - NailDriverSimulatorItem に limitCheckStartTime プロパティを追加した。
    
    吸着機能と同様の理由により追加した。
    デフォルトでは、開始から 0.3 秒までは限界判定を行わない。
  - 締結力判定処理を修正した。
    
    ビス方向の力で判定するように修正したつもりですが、もし考え方に間違いがありましたら改めて計算方法(式)を教えてください。
  - ビス打ち器が物体から離れたかどうかの判定の修正
    - nearCallback を呼ばない処理が続く回数のしきい値を設定するため、YAMLファイルに distantCheckCount を記述できるようにした。
    - 機器ごとの設定ではなく NailDriverSimulatorItem の方がよいかもしれませんが、その場合は修正します。

soumkin committed:
Snapshot at 0724a.
- Add new YAML parameter: distantCheckCount
- Fix header file dependency
- Add limit check start time
- Modify nailed object limit check

fkanehiro commented:
- シミュレーション開始後不安定というのは間違いないでしょうか。ロボットの動作がそれを引き起こしているだけではないのでしょうか。
- distantCheckCountはモデルに依存する話ではないので、SimulatorItemのプロパティにすべきではないでしょうか
- ビスの限界判定はビス方向の力で判定して下さい。現状はワールド座標系のX軸方向の力で判定されています。

soumkin committed:
Move distantCheckCount from YAML to property of SubSimulatorItem

soumkin commented:
> - _distantCheckCountはモデルに依存する話ではないので、SimulatorItemのプロパティにすべきではないでしょうか_

NailDriverSimulatorItem のプロパティとして設定するように変更し、YAML の設定を削除しました。(a0da8b5)

soum-takando commented:
> - ビスの限界判定はビス方向の力で判定して下さい。現状はワールド座標系のX軸方向の力で判定されています。

ビス方向の力の求め方は以下のような考え方でよいでしょうか?
- 1本目のビスを打つときに NailDriver の link()->R() \* nomal を計算する (nとする)
- 固定ジョイントに設定した dJointFeerback から得た f1 と n を使って n.dot(f) を計算する
- maxFasteningForce と比較する

fkanehiro commented:
はい、よろしいかと思います。

soumkin committed:
Update of snapshot at 0724a.
- modify nailed object limit check
- fix process of setting of barrel angle
- fix setKinematicStateToODE()
- remove debug messages

soum-takando commented:
j上記の計算方法を反映しました。
